coordinator/src/main/scala/filodb.coordinator/queryplanner/SinglePartitionPlanner.scala

```diff
@@ -115,5 +117,11 @@ class SinglePartitionPlanner(planners: Map[String, QueryPlanner],
     if (execPlans.size == 1) execPlans.head
     else PartKeysDistConcatExec(qContext, inProcessPlanDispatcher, execPlans)
   }
+
+  private def materializeTsCardinalities(logicalPlan: TsCardinalities, qContext: QueryContext): ExecPlan = {
+    val execPlans = planners.values.toList.distinct.map(_.materialize(logicalPlan, qContext))
+    if (execPlans.size == 1) execPlans.head
+    else TsCardReduceExec(qContext, inProcessPlanDispatcher, execPlans)
```

Comment:
Can you change the logic to not materialize with every planner and pick first ? We can just materialize with first planner. I realize you picked this pattern from existing code. You can change the others.

Reply:
Updated; I assert the size is 1 and just return the single exec plan from the planner that matched the "raw" filter. (See the thread between Amol and I on Slack for additional discussion).